compiler/src/dotty/tools/dotc/interactive/Completion.scala

```diff
@@ -386,6 +365,61 @@ object Completion {
         !name.isConstructorName && name.toTermName.info.kind == SimpleNameKind
       def isStable = true
     }
+
+    object CompletionScope {
+      def from(denotations: Seq[SingleDenotation])(using Context): CompletionScope = {
+        val mappings = denotations
+          .filter(den => include(den.symbol, den.name))
+          .toList
+          .groupBy(_.name)
+          .map((name, denots) => name.stripModuleClassSuffix -> denots)
+        CompletionScope(mappings)
+      }
+
+      def fromNamed(namedDenotations: Seq[(Name, SingleDenotation)])(using Context): CompletionScope = {
```

Comment:
From the type `Seq[(Name, SingleDenotation)]`, does it mean it's possible that from some tuple `(Name, SingleDenotation)`, the name and the `denot.name` is different?

Reply:
Yes, for renaming imports like `import Foo.{xxx => yyy}`